packages/name-controller/src/NameController.test.ts

```diff
@@ -2211,5 +2215,271 @@ describe('NameController', () => {
+
+        const controller = new NameController({
+          ...CONTROLLER_ARGS_MOCK,
+          providers: [provider1, provider2],
+        });
+
+        controller.state.names = {
+          [NameType.ETHEREUM_ADDRESS]: {
+            [`${VALUE_MOCK}1`]: {
+              [CHAIN_ID_MOCK]: {
+                name: null,
+                sourceId: null,
+                proposedNames: {
+                  [`${SOURCE_ID_MOCK}1`]: {
+                    proposedNames: ['NotExpiredName'],
+                    lastRequestTime: null,
+                    updateDelay: null,
+                  },
+                },
+              },
+            },
+            [VALUE_MOCK]: {
+              [CHAIN_ID_MOCK]: {
+                name: null,
+                sourceId: null,
+                proposedNames: {
+                  [`${SOURCE_ID_MOCK}1`]: {
+                    proposedNames: ['ExpiredName'],
+                    lastRequestTime:
+                      TIME_MOCK - PROPOSED_NAME_EXPIRE_DURATION - 1,
+                    updateDelay: null,
+                  },
+                  [`${SOURCE_ID_MOCK}2`]: {
+                    proposedNames: ['AnotherExpiredName'],
+                    lastRequestTime: null,
+                    updateDelay: null,
+                  },
+                },
+              },
+            },
+          },
+        };
+
+        await controller.updateProposedNames({
+          value: VALUE_MOCK,
+          type: NameType.ETHEREUM_ADDRESS,
+          variation: CHAIN_ID_MOCK,
+        });
+
+        expect(controller.state.names).toStrictEqual({
+          [NameType.ETHEREUM_ADDRESS]: {
+            [`${VALUE_MOCK}1`]: {
+              [CHAIN_ID_MOCK]: {
+                name: null,
+                sourceId: null,
+                proposedNames: {
+                  [`${SOURCE_ID_MOCK}1`]: {
+                    proposedNames: ['NotExpiredName'],
+                    lastRequestTime: null,
+                    updateDelay: null,
+                  },
+                },
+              },
+            },
+            [VALUE_MOCK]: {
+              [CHAIN_ID_MOCK]: {
+                name: null,
+                sourceId: null,
+                proposedNames: {
+                  [`${SOURCE_ID_MOCK}1`]: {
+                    proposedNames: [
+                      `${PROPOSED_NAME_MOCK}1`,
+                      `${PROPOSED_NAME_MOCK}1_2`,
+                    ],
+                    lastRequestTime: TIME_MOCK,
+                    updateDelay: null,
+                  },
+                  [`${SOURCE_ID_MOCK}2`]: {
+                    proposedNames: [
+                      `${PROPOSED_NAME_MOCK}2`,
+                      `${PROPOSED_NAME_MOCK}2_2`,
+                    ],
+                    lastRequestTime: TIME_MOCK,
+                    updateDelay: null,
+                  },
+                },
+              },
+            },
+          },
+        });
+      });
+    });
```

Comment:
As noted by Derek, if we're asserting a complex state object like this, we can cast it all to the relevant state type to validate it at compile time.

Reply:
Something like this? 
```
expect(controller.state.names).toStrictEqual({
          [NameType.ETHEREUM_ADDRESS]: {
            [`${VALUE_MOCK}1`]: {
            //.....//
            },
          },
        } as Record<NameType, Record<string, Record<string, NameEntry>>>);
```

That sounds like a nice idea. Should we want to apply partial assertions as well?

Reply:
Perhaps not as necessary for partials, but worth doing where we are asserting the full state.

No need to cast anymore as `toStrictEqual` accepts a template such as `.toStrictEqual<SomeType>({...})`.

Reply:
Thanks for explaining, that's a cool idea